### DISCUSSION TITLE
consider separating memory heaps into 2, thread-local , and global(process-local)(inter-thread) ,to improve semantics and perf.

### DISCUSSION THREAD

**Discussion by gintominto5329** [carbon-language/carbon-lang#3605] [Language design]
thanks.

**Comment by chandlerc** [comment]
While I think the ideas around separate heaps are interesting and definitely worth exploring, it's a huge topic.

A suggestion in the discussion forums without someone interested in digging inot the details of what this would look like, how it would work, and especially how it would impact interop with C++, I'm not sure it makes sense to have a discussion thread.

This also isn't well aligned with our current [roadmap] for 2024, so I'm going to go ahead and close this discussion. If this language design area is something you're particularly interested in exploring, I would suggest contributing to some other aspects of the Carbon project, especially in line with our roadmap or specific gaps in the design that we will need to close getting to our [0.1 milestone]. And then it may make more sense to start a longer discussion of the full scope of what this would entail and how it would work.

[roadmap]: docs/project/roadmap.md
[0.1 milestone]: docs/project/milestones.md#milestone-01-a-minimum-viable-product-mvp-for-evaluation
